frontend/components/instruments/SpatialInstrumentInternal.js

```diff
@@ -0,0 +1,180 @@
+import React, {useState, useRef, useEffect} from "react";
+import PropTypes from "prop-types";
+import SoundPoint from "./SoundPoint";
+import SamplePlayer from "./SamplePlayer";
+import STYLES from "./SpatialInstrument.module.scss";
+
+/* TODO: there is a bug in which the player crashes
+    not sure how to reproduce.
+    Probably has something to do with toggling based on maxDist.
+    "Uncaught DOMException: Failed to execute 'start' on 'AudioBufferSourceNode':
+    cannot call start more than once."
+    Caused when on border between two sounds based on maxDist.
+    *
+    Also crashes when using Pad instrument at same time.
+    probably caused by reusing AudioBufferSourceNode - retriggering sound SamplePlayer before
+     the old AudioBufferSourceNode is thrown away.
+     https://developer.mozilla.org/en-US/docs/Web/API/AudioBufferSourceNode
+   fixed by keeping everything playing at all times, just using 0 volume to "turn off" nodes.
+*/
+
+/*
```

Comment:
I like the idea here of having a drop-in svg rendering component, but I'd warn against trying to make a generalized system before the specific one. What's already put together works great, and it's being used in a single specific context, so we actually don't need the hook for this override, so my suggestion would be to cut it, since as currently implemented it's a bit of extra code but doesn't actually implement any functionality.